vendor: update buildkit to v0.18.0-rc1

tonistiigi opened the pull request:
- [x] depends on https://github.com/moby/moby/pull/48925
- [x] depends on https://github.com/moby/moby/pull/48932
- [x] depends on https://github.com/moby/buildkit/pull/5547


**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section:
-->
```markdown changelog
TODO: add description for changelog
```

tonistiigi committed:
vendor: update buildkit to v0.18.0-rc1

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

crazy-max commented:
@tonistiigi As discussed offline about https://github.com/moby/moby/actions/runs/11967527151/job/33364978457#step:13:43433

```
 === FAIL: client TestIntegration/TestDiffAsParentMultipleLayers/worker=dockerd-containerd (2.09s)
    client_test.go:7857: checkAllReleasable: skipping check for exported tars in non-containerd test
    client_test.go:7855: 
        	Error Trace:	/src/client/client_test.go:7855
        	            				/src/client/mergediff_test.go:1454
        	            				/src/client/client_test.go:7721
        	            				/src/client/mergediff_test.go:1281
        	            				/src/util/testutil/integration/run.go:211
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 5
        	Test:       	TestIntegration/TestDiffAsParentMultipleLayers/worker=dockerd-containerd
```

This relates to https://github.com/moby/buildkit/pull/5516/files#diff-57cdc3e29d0ef725a50897a4fbd1c4f76353429fa66d3a785dcbc17dcd7b6f31R7855

Which needs a change on buildkit repo to check we are using containerd snap. I'm looking at it.


It needs a change in buildkit

crazy-max commented:
Tested with https://github.com/moby/moby/pull/48924 and looks good.

thaJeztah committed:
vendor: github.com/moby/buildkit 94d0f6ed85e5 (master / v0.18.0-rc.2)

full diff: https://github.com/moby/buildkit/compare/v0.18.0-rc1...94d0f6ed85e504e1851495fa98485ffbd9cfddb0

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah reviewed (approved):
LGTM
